nixos/modules/services/networking/frp-client.nix

```
@@ -0,0 +1,70 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.frp-client;
+  settingsFormat = pkgs.formats.ini { };
+  configFile = settingsFormat.generate "frpc.ini" cfg.settings;
+in
+{
+  options = {
+    services.frp-client = {
+      enable = mkEnableOption (lib.mdDoc "Frp client configuration");
+
+      package = mkPackageOptionMD pkgs "frp" { };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = { };
+        description = lib.mdDoc ''
+          Frp client configuration, for configuration options see the example on [github](https://github.com/fatedier/frp/blob/dev/conf/frpc_full.ini)
+        '';
+        example = literalExpression ''
+          {
+            common = {
+              server_addr = "x.x.x.x";
+              server_port = 7000;
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services = {
+      frp-client = {
+        wants = [ "network-online.target" ];
```

`man systemd.special`:

> Units that strictly require a configured network connection should pull in network-online.target (via a Wants= type dependency) and order themselves after it.

```suggestion
        wants = [ "network-online.target" ];
        after = [ "network-online.target" ];
```

---

Thank you. I've updated it. Later I will rework this PR to combine two frp modules.